types/services.d.ts

```diff
@@ -269,11 +264,11 @@ declare namespace Realm {
             /**
              * A base change event containing the properties which apply across operation types.
              */
-            type BaseChangeEvent = {
+            type BaseChangeEvent<T extends OperationType> = {
```

Comment:
Maybe `s/T/OpType/` so that it is clear that it isn't a type (in the usual sense) but one of the string values?